### PR TITLE
Allow to ignore cloned lines field on duplication test

### PR DESCRIPTION
**Pull request by josemiguelmelo**

### CODE REVIEW

**Comment by mrfyda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/scala/codacy/plugins/test/duplication/CheckstyleFormatParser.scala  1
- src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala  1
         

Clones removed
==============
+ src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codacy/codacy-plugins-test/pullRequest?prid=4894691&bid=16208195)

**Comment by lolgab** [comment]
I would say to find something more in line with the checkstyle standard. I don't like the property on the `checkstyle` tag.

**Comment by mrfyda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/scala/codacy/plugins/test/duplication/CheckstyleFormatParser.scala  1
- src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala  1
         

Clones removed
==============
+ src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codacy/codacy-plugins-test/pullRequest?prid=4894691&bid=16208195)

**Comment by mrfyda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/scala/codacy/plugins/test/duplication/CheckstyleFormatParser.scala  1
- src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala  1
         

Clones removed
==============
+ src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codacy/codacy-plugins-test/pullRequest?prid=4894691&bid=16208195)

File: src/main/scala/codacy/plugins/test/checkstyle/CheckstyleImplicits.scala [hunk 1]
```diff
@@ -12,6 +12,14 @@ object CheckstyleImplicits {
       } yield propertyNode \@ "value"
     }
 
+    def isPropertyDefined(name: String): Boolean = {
```
Comment (lolgab) [inline src/main/scala/codacy/plugins/test/checkstyle/CheckstyleImplicits.scala]: You can simplify like this:
```scala
def existsProperty(name: String): Boolean = {
  (node \ "property").exists(p => (p \@ "name") == name)
}
```

File: src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala [hunk 2]
```diff
@@ -33,16 +33,33 @@ object DuplicationTests extends ITest {
...
+  private def ignoreClonedLines(res: Try[Set[DuplicationClone]]): Try[Set[DuplicationClone]] = {
+    Try {
+      res match {
+        case Success(resultsList) =>
+          resultsList.map { r =>
+            r.copy(cloneLines = "")
+          }
+        case Failure(err) => throw new Exception(err)
+      }
+    }
```
Comment (lolgab) [inline src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala]: Here you can do 
```scala
res.map(_.map(_.copy(cloneLines = "")))
```

File: src/main/scala/codacy/plugins/test/checkstyle/CheckstyleImplicits.scala (hunk 1 above)
Comment (josemiguelmelo) [reply src/main/scala/codacy/plugins/test/checkstyle/CheckstyleImplicits.scala]: Nice one, done ✅

File: src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala (hunk 2 above)
Comment (josemiguelmelo) [reply src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala]: Changed it 👌

**Comment by mrfyda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/scala/codacy/plugins/test/duplication/CheckstyleFormatParser.scala  1
         

Clones removed
==============
+ src/main/scala/codacy/plugins/test/duplication/DuplicationTests.scala  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codacy/codacy-plugins-test/pullRequest?prid=4894691&bid=16208195)

**Pull request merged**